### PR TITLE
[MB-6219] Bring support yaml in sync with Prime 

### PR DESCRIPTION
**Pull request by shimonacarvalho**
## Description

Found some broken things that needed to be fixed to make load testing prime workflow succeed.
Needed to bring mtoShipments and mtoAgent code in support.yaml up to date with prime.yaml (hate that this is necessary)
Basically copy pasted from prime to update the following:

* Updated MTO Shipment support model/payload to match prime version
* Updated MTO Agent yaml and model/payload funcs to match prime.yaml
* Updated UpdatePostCounseling info to return entitlement, which was in the redoc but not being populated

## Setup

Test the support/updateMTOShipmentStatus and prime/UpdatePostCounselingInfo endpoint to check that they work and return all info expected in the redoc. 

## References

* [Jira story](https://dp3.atlassian.net/browse/mb-6219) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

File: swagger/prime.yaml [hunk 1]
```diff
@@ -20,6 +20,13 @@ tags:
   - name: mtoShipment
   - name: paymentRequest
   - name: mtoServiceItem
+x-tagGroups:
```
Comment (shimonacarvalho) [inline swagger/prime.yaml:23]: This is just a nice way to group the endpoints under a heading. I was trying it in preparation for adding some more documentation under a different heading. Can remove from this pr if desired, it is just cosmetic.

File: pkg/handlers/primeapi/payloads/model_to_payload.go [hunk 2]
```diff
@@ -332,7 +332,7 @@ func PaymentServiceItemParams(paymentServiceItemParams *models.PaymentServiceIte
 	return &payload
 }
 
-// MTOShipment payload
+// MTOShipment converts MTOShipment model to payload
```
Comment (sandy-wright) [inline pkg/handlers/primeapi/payloads/model_to_payload.go:335]: thank you for these comment updates ✨

File: pkg/services/move_task_order/move_task_order_updater.go [hunk 3]
```diff
@@ -132,9 +132,10 @@ type UpdateMoveTaskOrderQueryBuilder interface {
 func (o *moveTaskOrderUpdater) UpdatePostCounselingInfo(moveTaskOrderID uuid.UUID, body movetaskorderops.UpdateMTOPostCounselingInformationBody, eTag string) (*models.Move, error) {
 	var moveTaskOrder models.Move
 
-	err := o.db.Q().Eager(
+	err := o.db.Q().EagerPreload(
```
Comment (sandy-wright) [inline pkg/services/move_task_order/move_task_order_updater.go:135]: yesss

**Review by eamahanna — APPROVED** [review]
Verified all the fields are correct and match redoc

File: pkg/handlers/supportapi/internal/payloads/model_to_payload.go [hunk 4]
```diff
@@ -211,42 +212,77 @@ func Address(address *models.Address) *supportmessages.Address {
...
 
-// MTOServiceItem payload
+// MTOShipments converts an array of MTOShipment models to a payload
+func MTOShipments(mtoShipments *models.MTOShipments) *supportmessages.MTOShipments {
+	payload := make(supportmessages.MTOShipments, len(*mtoShipments))
+
+	for i, m := range *mtoShipments {
+		copyOfM := m // Make copy to avoid implicit memory aliasing of items from a range statement.
+		payload[i] = MTOShipment(&copyOfM)
+	}
```
Comment (sandy-wright) [inline pkg/handlers/supportapi/internal/payloads/model_to_payload.go:281]: 💯

**Review by sandy-wright — APPROVED** [review]
LGTM! Some manual testing, and ran it through the non-sequential load tests 👍

**Pull request merged**